RecDP/pyrecdp/primitives/operations/text_split.py

```diff
@@ -1,46 +1,108 @@
+        return text_splitter.split_text(text)
+
     return process
 
-class DocumentSplit(BaseLLMOperation):
-    def __init__(self, text_key = 'text', inplace = True, language = 'en'):
-        settings = {'text_key': text_key, 'inplace': inplace, 'language': language}
+
+class BaseDocumentSplit(BaseLLMOperation, ABC):
+    def __init__(self, text_key: str,
+                 inplace: bool = False,
+                 args_dict: Optional[Dict] = {}):
+
+        settings = {
+            'inplace': inplace,
+            'text_key': text_key
+        }
+        settings.update(args_dict)
         super().__init__(settings)
         self.text_key = text_key
-        self.inplace = inplace
-        self.language = language
-        self.actual_func = None
-        self.support_spark = True
         self.support_ray = True
-        
-    def process_rayds(self, ds: Dataset) -> Dataset:
-        if self.inplace:
-            new_name = self.text_key
+        self.support_spark = True
+        self.text_splitter = self._get_text_splitter()
+        if inplace:
+            self.split_text_column = self.text_key
         else:
-            new_name = 'split_text'
-        if self.actual_func is None:
-            self.actual_func = prepare_func_sentencesplit(lang = self.language)
-        return ds.map(lambda x: self.process_row(x, self.text_key, new_name, self.actual_func))
-    
-    def process_spark(self, spark, spark_df: DataFrame) -> DataFrame:
+            self.split_text_column = 'split_text'
+
+        self.text_split_func = prepare_func_text_split(self.text_splitter)
+
+    def _get_text_splitter(self) -> TextSplitter:
+        """base interface to get text splitter"""
+
+    def process_rayds(self, ds: Dataset = None):
+        def split_text(sample, text_split_func) -> List[Dict[str, Any]]:
+            result = []
+            for text in text_split_func(sample[self.text_key]):
+                row = dict(**sample)
+                row[self.split_text_column] = text
+                result.append(row)
+            return result
+
+        return ds.flat_map(lambda sample: split_text(sample, self.text_split_func))
+
+    def process_spark(self, spark, ds: DataFrame = None):
         import pyspark.sql.functions as F
         from pyspark.sql import types as T
-        if self.inplace:
-            new_name = self.text_key
-        else:
-            new_name = 'split_text'
-        sentencesplit_udf = F.udf(prepare_func_sentencesplit(lang = self.language))
-        return spark_df.withColumn(new_name, sentencesplit_udf(F.col(self.text_key)))
-    
-LLMOPERATORS.register(DocumentSplit)
\ No newline at end of file
+
+        split_text_udf = F.udf(self.text_split_func, T.ArrayType(T.StringType()))
+        ds = ds.withColumn(self.split_text_column, split_text_udf(F.col(self.text_key)))
+        ds = ds.withColumn(self.split_text_column, F.explode(F.col(self.split_text_column)))
+        return ds
+
+
+class DocumentSplit(BaseDocumentSplit):
```

Comment:
why some of the codes are put under llmutils/document/splitter.py and some are put here?